bindings/wasm/src/account/wasm_account.rs

```diff
@@ -0,0 +1,365 @@
+    &self,
+    fragment: String,
+    document: &WasmDocument,
+    signature_options: WasmSignatureOptions,
+  ) -> PromiseDocument {
+    let account = self.0.clone();
+    let mut doc: IotaDocument = document.0.clone();
+    let options: SignatureOptions = SignatureOptions::from(signature_options);
+
+    let promise: Promise = future_to_promise(async move {
+      account
+        .as_ref()
+        .borrow_mut()
+        .sign(fragment.as_str(), &mut doc, options)
+        .await
+        .map(|_| JsValue::undefined())
+        .wasm_result()?;
+      JsValue::from_serde(&doc).wasm_result()
+    });
+    promise.unchecked_into::<PromiseDocument>()
+  }
+
+  /// Signs a {@link Presentation} the key specified by `fragment`.
+  #[wasm_bindgen(js_name = createSignedPresentation)]
+  pub fn create_signed_presentation(
+    &self,
+    fragment: String,
+    presentation: &WasmPresentation,
+    signature_options: WasmSignatureOptions,
+  ) -> PromisePresentation {
+    let account = self.0.clone();
+    let mut pres: Presentation = presentation.0.clone();
+    let options: SignatureOptions = SignatureOptions::from(signature_options);
+
+    wasm_logger::init(wasm_logger::Config::default());
+    log::info!("{:?}", options);
+
+    let promise: Promise = future_to_promise(async move {
+      account
+        .as_ref()
+        .borrow_mut()
+        .sign(fragment.as_str(), &mut pres, options)
+        .await
+        .map(|_| JsValue::undefined())
+        .wasm_result()?;
+      JsValue::from_serde(&pres).wasm_result()
+    });
+    promise.unchecked_into::<PromisePresentation>()
+  }
+
+  /// Signs arbitrary `data` with the key specified by `fragment`.
+  #[wasm_bindgen(js_name = createSignedData)]
+  pub fn create_signed_data(
+    &self,
+    fragment: String,
+    data: &JsValue,
+    signature_options: WasmSignatureOptions,
+  ) -> Result<Promise> {
+    let account = self.0.clone();
+    let mut verifiable_properties: VerifiableProperties = data.into_serde().wasm_result()?;
+    let options: SignatureOptions = SignatureOptions::from(signature_options);
+
+    let promise = future_to_promise(async move {
+      account
+        .as_ref()
+        .borrow_mut()
+        .sign(fragment.as_str(), &mut verifiable_properties, options)
+        .await
+        .map(|_| JsValue::undefined())
+        .wasm_result()?;
+      JsValue::from_serde(&verifiable_properties).wasm_result()
+    });
+    Ok(promise)
+  }
+
+  /// Overwrites the {@link Document} this account manages, **without doing any validation**.
+  ///
+  /// # WARNING
+  ///
+  /// This method is dangerous and can easily corrupt the internal state,
+  /// potentially making the identity unusable. Only call this if you fully
+  /// understand the implications!
+  #[wasm_bindgen(js_name = updateDocumentUnchecked)]
+  pub fn update_document_unchecked(&mut self, document: WasmDocument) -> Promise {
+    let account = self.0.clone();
+    future_to_promise(async move {
```

Comment:
I think this should take a reference to document and clone it to prevent unexpectedly nulling its pointer in Javascript.